File: 2025-09-10/arrayDiff.js
function arrayDiff(arr1, arr2) {
	
  return arr1
    .filter(element1 => !arr2.some(element2 => element1 === element2))
    .concat(arr2.filter(element2 => !arr1.some(element1 => element2 === element1)))
    .sort();
}

console.log(arrayDiff(["apple", "banana"], ["apple", "banana", "cherry"]));
console.log(arrayDiff(["apple", "banana", "cherry"], ["apple", "banana"]));
console.log(arrayDiff(["one", "two", "three", "four", "six"], ["one", "three", "eight"]));
console.log(arrayDiff(["two", "four", "five", "eight"], ["one", "two", "three", "four", "seven", "eight"]));
console.log(arrayDiff(["I like freeCodeCamp"], ["I like rocks"]));

File: 2025-09/2025-09-10/array_diff.py
def array_diff(arr1, arr2):

    return sorted( \
    [element1 for element1 in arr1 if not any([element1 in arr2])] + \
    [element2 for element2 in arr2 if not any([element2 in arr1])] \
    )

print(array_diff(["apple", "banana"], ["apple", "banana", "cherry"]))
print(array_diff(["apple", "banana", "cherry"], ["apple", "banana"]))
print(array_diff(["one", "two", "three", "four", "six"], ["one", "three", "eight"]))
print(array_diff(["two", "four", "five", "eight"], ["one", "two", "three", "four", "seven", "eight"]))
print(array_diff(["I like freeCodeCamp"], ["I like rocks"]))